nemo/collections/asr/models/aed_multitask_models.py

```diff
@@ -698,9 +703,54 @@ def forward(
 
         return transf_log_probs, encoded_len, enc_states, enc_mask
 
+    # Wrapper function for updating output dict with metrics.
+    # Performs single update, reset loop with metric functions.
+    # Expand as needed.
+    def _eval(
+        self,
+        output_dict: Dict[str, torch.Tensor],
+        batch: PromptedAudioToTextMiniBatch,
+        encoded_states: torch.Tensor,
+        encoded_len: torch.Tensor,
+        encoded_mask: torch.Tensor,
+        eval_prefix: Literal["val", "test", "training_batch"],
+        return_all_metrics: bool,
+    ):
+        self.wer.update(
+            predictions=encoded_states,
+            predictions_lengths=encoded_len,
+            targets=batch.transcript,
+            targets_lengths=batch.transcript_lens,
+            predictions_mask=encoded_mask,
+            input_ids=batch.prompt,
+        )
+        # TODO: Remove conditional once wer reflects bleu batch behavior.
+        wer, wer_num, wer_denom = self.wer.compute()
+        if return_all_metrics:
+            output_dict.update(
+                {f"{eval_prefix}_wer": wer, f"{eval_prefix}_wer_num": wer_num, f"{eval_prefix}_wer_denom": wer_denom}
+            )
+        else:
+            output_dict.update({f"{eval_prefix}_wer": wer})
+        self.wer.reset()
+
+        self.bleu.update(
+            predictions=encoded_states,
+            predictions_lengths=encoded_len,
+            targets=batch.transcript,
+            targets_lengths=batch.transcript_lens,
+            predictions_mask=encoded_mask,
+            input_ids=batch.prompt,
+            langs=[c.custom["target_lang"] for c in flatten_mixed(batch.cuts)],
```

Comment:
I have been thinking about this as well. Currently we compute wer and bleu score irrespective of task. I thought about this and added the change based on tokenizer type. As this information can be grabbed from prompt and this `target_lang` key might change later based on tokenizer type (canary,canar2). See this branch code: https://github.com/NVIDIA/NeMo/blob/selectively_run_metrics/nemo/collections/asr/models/aed_multitask_models.py#L775-L831 

So instead adding this to cut we can grab from prompt which can be added to tokenizer type class itself.

Reply:
Oh I like that, reduces the amount of logging. I'll add this to the PR.

I also spoke with piotr about this offline and we're just going to create a new attribute called 'bleu_token' that you can specify in the config. Less mucking about with prompting.

Reply:
Actually, let's add this code in a separate PR. I want to add a wrapper around it so it can extend to new evaluation cases as they come up (speaker asr and the like) and that's going to add more locs on this pr.

Reply:
@nithinraok check the new `multitask` metric class. The functions at the bottom should pull off the behavior now.